doc/source/conf.py

```diff
@@ -61,13 +62,33 @@
 autodoc_typehints_description_target = "documented"
 autodoc_member_order = "bysource"
 
+from packaging.version import parse as parse_version
```

Comment:
I think I agree that this is a lot of additional complexity for minimal benefit. I suggest we always link to version/stable, and accept that in some possible scenarios people might end up looking at documentation for authentication schemes they can't use. We have no plans to radically change the authentication in a backwards-incompatible way though, so I think it is unlikely that this scenario will come to pass.

This extra code is just something else that could cause us problems in the future.